### ISSUE TITLE
net: only js/wasm returns an error on writes to closed connections

### ISSUE THREAD

**Issue by FiloSottile** [golang/go#28650]
If a localhost connection is closed by the peer, all other GOOS seem to allow writes to happen without error, while js/wasm returns `write tcp 127.0.0.1:XXX->127.0.0.1:1: Socket is not connected` with XXX changing.

I admit it does feel like something that should return an error, and I don't know if there is a precise requirement to do otherwise, but js being the only GOOS to do it is very inconsistent, and not even NaCl on the playground with its fictitious network returns an error.

This came up because js/wasm were the only broken TryBots in some crypto/tls CL: https://storage.googleapis.com/go-build-log/ca02321c/js-wasm_d63aa3d8.log

https://play.golang.org/p/lRhq44NJXTr (this specific program will actually panic on js/wasm due to #28649, but it proves the `log.Print` is being reached)

/cc @neelance

**Comment by mikioh** [comment]
Unfortunately, the fake implementation of TCP-like byte-sequence full-duplex pipe on JS is broken. Also, it doubles the maintenance cost by having each implementation on NaCl and JS. I have a CL that may solve the issues: https://go-review.googlesource.com/c/go/+/120958, but it will probably land in Go 1.13 or above.

Can you please skip the roadblock test cases for now?

**Comment by FiloSottile** [comment]
Yeah I adapted the tests, so not urgent on my side. Feel free to re-milestone as you wish.

What is blocking that CL?

**Comment by mikioh** [comment]
> What is blocking that CL?

Just for congestion control; the plan for Go 1.12 had AIX and Fuchsia ports that need to churn up the runtime, os and net packages. Like other communication protocols, congestion control is reviewee/sender side's responsibility (and flow control is reviewer/receiver side's responsibility.)
